calebrotich/Fast-Food
# Develop

**What Does this PR do?**
Adds end-points for fast-food-fast application

**Description of task to be completed**
Add endpoints for the api
1. GET all orders
2. GET specific order
3. POST an order
4. UPDATE order

**How should this be Manually tested?**
Clone the repository 
`git clone -b develop git@github.com:calebrotich10/Fast-Food.git`
`cd Fast-Food`

*setup virtual environment*
`virtualenv env`
`Activate the virtual environment`

*Install dependencies*
`pip install -r requirements.txt`

*Functionalities*
GET `/orders` Get all the orders.
GET `/orders/<id>`  Fetch a specific order
POST  `/orders` Place a new order.
PUT `/orders/<id>/<status>` Update the status of an order.

*What are the relevant pivotal tracker stories?*
1. [User Should be able to update order status](https://www.pivotaltracker.com/story/show/160262735)
2. [User should be able to place a new order for food](https://www.pivotaltracker.com/story/show/160261179)
3. [User should be able to fetch a specific order](https://www.pivotaltracker.com/story/show/160262386)
4. [User should be able to get a list of orders](https://www.pivotaltracker.com/story/show/160262021)
5. [User should have a data store for the orders](https://www.pivotaltracker.com/story/show/160266978)
6. [Test if user can fetch all orders](https://www.pivotaltracker.com/story/show/160266646)
7. [Test if user can fetch a specific order](https://www.pivotaltracker.com/story/show/160266407)
8. [Test if user can place an order](https://www.pivotaltracker.com/story/show/160264993)
9. [Test if user can update order status](https://www.pivotaltracker.com/story/show/160266014)
10. [Developer should be able to see a coverage test report](https://www.pivotaltracker.com/story/show/160344771)

*Postman tests screenshots*
<i>1. Update Order Status</i>
![screenshot 132](https://user-images.githubusercontent.com/29451552/45255539-90c08a00-b390-11e8-893b-6b579c8e6549.png)

<i>2. Add a new order</i>
![screenshot 131](https://user-images.githubusercontent.com/29451552/45255540-928a4d80-b390-11e8-9bae-fb286ba68971.png)

<i>3. Get a specific Order</i>
![screenshot 130](https://user-images.githubusercontent.com/29451552/45255541-93bb7a80-b390-11e8-80f9-1a5df59061dd.png)

<i>4. GET all orders</i>
![screenshot 129](https://user-images.githubusercontent.com/29451552/45255542-95853e00-b390-11e8-91dd-11b6e527b05b.png)
